library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -1056,86 +1202,118 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
         }
         mMinZoom = minZoom
         mMinZoomMode = type
-        if (zoom <= resolveZoom(minZoom, type)) {
-            zoomTo(resolveZoom(minZoom, type), true)
+        if (zoom <= getMinZoom()) {
+            zoomTo(getMinZoom(), animate = true)
         }
     }
 
     //endregion
 
     //region Apply values
 
+    private val mCancelAnimationListener = object : Animator.AnimatorListener {
```

Comment:
Do you think we should setState(NONE) onAnimationCancel also?

Reply:
Well the animation is only cancelled when `mClearAnimation` is set to `true` which is done in `setState(NONE)` ... so currently it's the other way around - the state change causes the animationListener to trigger onCancel. It wouldn't hurt though I guess since it will already be in that state.

Reply:
Well it's not important then, as you wish